src/attributes.jl

```diff
@@ -1856,6 +1856,86 @@ function attribute_value_type(::ConstraintConflictStatus)
     return ConflictParticipationStatusCode
 end
 
+"""
+    UserDefinedFunction(name::Symbol, arity::Int) <: AbstractModelAttribute
+
+Set a user-defined function by the name of `name` with `arity` arguments.
+
+The value to be set is a tuple containg one to three functions, representing the
+zero, first, and second-order derivatives of the function.
+
+For uni-variate functions with `arity == 1`, the functions must have the form:
+
+ * `f(x::T)::T`: returns the evaluation of the function at `x`
```

Comment:
```suggestion
 * `f(x::T)::T`: returns the value of the function at `x`
```
not a strong opinion though